### PR TITLE
CLE error modal clean up

### PR DESCRIPTION
**Pull request by minhaminha**
## Product Description

For both Case List and Case List Explorer, it will now show an error banner instead of a modal if user input is too long. User input has also been further limited to match the max URL length. 

## Technical Summary

[Jira](https://dimagi-dev.atlassian.net/browse/QA-5141)

While working on URL obfuscation for CLE, I introduced a character count limit to the user input textbox but because it wasn't the focus of that PR,  I did it a little sloppily - this PR fixes that.

This PR also fixes a previously existing bug that turned up while QA testing, where the report would continue showing the loading spinner if you cancelled out of the error modal. 

## Feature Flag

Case List Explorer

### Safety story

Tested locally and on staging.

### QA Plan

QA Ticket is the same as the Jira link above. 

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/reports.async.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-hqDefine("reports/js/reports.async", function () {
+hqDefine("reports/js/reports.async", ['hqwebapp/js/alert_user'], function (alertUser) {
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/reports.async.js:1]: nit whitespace to match existing conventions

```suggestion
hqDefine("reports/js/reports.async", [
    'hqwebapp/js/alert_user'
], function (alertUser) {
```

File: corehq/apps/reports/static/reports/js/reports.async.js [hunk 2]
```diff
@@ -240,12 +235,10 @@ hqDefine("reports/js/reports.async", function () {
...
         self.loadingIssueModal.on('hide hide.bs.modal', function () {
-            if (self.issueAttempts > 0) {
-                self.hqLoading = $(self.loaderClass);
-                self.hqLoading.find('.js-loading-spinner').addClass('hide');
-                self.hqLoading.find('h4').text(gettext('We were unsuccessful loading the report:'))
-                    .attr('style', 'margin-bottom: 10px;');
-            }
+            self.hqLoading = $(self.loaderClass);
+            self.hqLoading.find('.js-loading-spinner').addClass('hide');
+            self.hqLoading.find('h4').text(gettext('We were unsuccessful loading the report:'))
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/reports.async.js:240]:
```suggestion
            self.hqLoading.find('h4').text(gettext('We were unsuccessful loading the report:'));
```

Comment (minhaminha) [reply corehq/apps/reports/static/reports/js/reports.async.js]: hmm I wonder why stickler isn't picking up on these kinds of things anymore

Comment (minhaminha) [reply corehq/apps/reports/static/reports/js/reports.async.js:240]: whoops that's because this isn't the end of the line

Comment (biyeun) [reply corehq/apps/reports/static/reports/js/reports.async.js:240]: ahhhhh

**Review by biyeun — APPROVED** [review]

**Pull request merged**